src/t_zset.c

```diff
@@ -2253,8 +2253,7 @@ static void zdiffAlgorithm1(zsetopsrc *src, long setnum, zset *dstzset, size_t *
         for (j = 1; j < setnum; j++) {
             /* It is not safe to access the zset we are
                 * iterating, so explicitly check for equal object. */
-            if (src[j].subject == src[0].subject ||
-                    zuiFind(&src[j],&zval,&value)) {
+            if (zuiFind(&src[j],&zval,&value)) {
```

Comment:
you should update the comment.
options:
1. remove the comment that's no longer needed.
2. keep the comment and the code (which may not be needed, but doesn't harm either (maybe something will some day change and this check will save the day).
3. remove the check as you did, and update the comment to state that it's unsafe but will never happen due to the check in `zsetChooseDiffAlgorithm`

either way:
1. you're welcome to add this optimization to t_set.c
2. please add a test to cover that (unless you already did) 8-)

Reply:
ohh, i see you did remove the comment (reviewed the commits one by one, and spoke too soon).
well, maybe you wanna reconsider based on my feedback. (this can be risky)

Reply:
This definitely requires a comment stating the unsafeness (if it remains without the check), I've missed that, thanks por pointing it out.

Another option I thought about was just adding a simple loop checking for this just before the qsort call. I know this may be some "over-optimization thinking" of my part, but it just bothers me to have that "useless" check in every inner iteration of the algorithm. 😅 

But I know full well that it probably doesn't have a meaningful impact in the execution time, so if you think it's better to leave it, no problem, I can change it back!

Reply:
All options are OK, I guess the safest one is to add back the check, with a comment that explains why it was needed, and even say that it's not currently needed due to the outer check.

Reply:
Done!